### PR TITLE
Queue deduplication and collection updates

### PR DESCRIPTION
**Pull request by dgasmith**
## Description
 - Refactors collection constructors to allow initialize from a base and from a complete structure.
 - Compute now prevents duplicate and will return if a job was submitted or if was complete/exists in the queue.
 - Starts a BioFragment entry (computes torsions and will change based on OpenFF requirements)
 - TorsionDrives now handle duplicate procedures

Overall this PR faced serious feature creep as additional issues were discovered. This is deep enough we should pull it in soon and start building ontop of these changes.

## Status
- [x] Ready to go

### CODE REVIEW

**Comment by dgasmith** [comment]
This pull request **introduces 5 alerts** when merging cb8c113ecbecf498c33b39a96c7fcd3176096738 into ba2fb263054e9095c7a7c99b5065acefcd56c7ce - [view on LGTM.com](https://lgtm.com/projects/g/MolSSI/QCFractal/rev/pr-63fc4c1da49e9530da0b14a2c0dd8f5018db7f79)

**new alerts:**

* 3 for Unused local variable
* 2 for Unused import

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by dgasmith** [comment]
This pull request **introduces 2 alerts** when merging ef1e34dfba6aba403caecc1991b58da287761b1f into ba2fb263054e9095c7a7c99b5065acefcd56c7ce - [view on LGTM.com](https://lgtm.com/projects/g/MolSSI/QCFractal/rev/pr-f62daa529c8efa3399512fe554963eb704bb9389)

**new alerts:**

* 2 for Unused local variable

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: qcfractal/interface/client.py [hunk 1]
```diff
@@ -273,12 +276,12 @@ def get_results(self, **kwargs):
         else:
             return r.json()["data"]
 
-    def get_procedures(self, procedure_id, **kwargs):
+    def get_procedures(self, procedure_id, return_objects=True):
```
Comment (Lnaden) [inline qcfractal/interface/client.py]: Will there ever be a reason to provide other KWargs?

**Comment by Lnaden** [comment]
I restarted the failed Travis. It was an HTTP timeout on their end, so it should be fine

File: qcfractal/interface/client.py (hunk 1 above)
Comment (dgasmith) [reply qcfractal/interface/client.py]: I don't think there will be a reason for general `**kwargs` here, but there may be additional `kwargs`.

**Pull request merged**